### PR TITLE
GE-959: role assignment and revocation permission decorator

### PR DESCRIPTION
**Pull request by cobusc**
Added decorator that will check whether the user that wants to assign or remove a role on a domain or site to another user, has that role. Example:
```
@requester_has_role(body_field=1)
def usersiterole_create(request, body, **kwargs):
    ...
```
The body must contain the `user_id` of the recipient, the `role_id` of the role that must be assigned and the `site_id` of the site for which the user will must get the role.

```
@requester_has_role(target_user_id_field=1, site_id_field=2, role_id_field=3)
def usersiterole_delete(request, user_id, site_id, role_id, **kwargs):
      ...
```
Advanced:
```
@requester_has_role(request_field=1, body_field=0, target_user_id_field="some_user",
                    site_id_field="some_site", role_id_field="some_role")
def call_with_body(body, request, **kwargs):  # Note that request is not first
    return True
```

### CODE REVIEW

File: management_layer/tests/test_permission_utils.py [hunk 1]
```diff
@@ -341,56 +326,290 @@ def multiple_requirements(_request):
...
+            self, mocked_get_user_roles_for_domain):
+        """
+        If a body field is present, it must be a dictionary containing:
+        * the target user id (user_id)
+        * the role to be assigned (role_id)
+        * either the site or domain to which it must be assigned (site_id or role_id)
+        """
+        @requester_has_role(request_field=1, body_field=0, target_user_id_field="some_user",
+                            domain_id_field="some_domain", role_id_field="some_role")
+        def call_with_body(body, request, **kwargs):  # Note that request is not first
```
Comment (ruands) [inline management_layer/tests/test_permission_utils.py]: Out of curiosity, why the swap?

**Comment by ruands** [comment]
+1, just one question :)

File: management_layer/tests/test_permission_utils.py (hunk 1 above)
Comment (cobusc) [reply management_layer/tests/test_permission_utils.py]: @ruands To illustrate that the `request_field` and `body_field` parameters are correctly used to identify the corresponding parameters.

Comment (ruands) [reply management_layer/tests/test_permission_utils.py]: I see, thanks 👍

**Pull request merged**